__tests__/base.js

```diff
@@ -1147,6 +1147,47 @@ function runBaseTest(name, useProxies, autoFreeze, useListener) {
 			expect(isDraft(array)).toBeFalsy()
 		})
 	})
+
+	describe('class with getters and methods', () => {
+		class State {
+			constructor() {
+				this[immerable] = true
+				this.word1 = "bar"
+				this.word2 = "foo"
+				this.foo = 0
+				this._bar = {baz: 1}
+			}
+			get chars() {
+				return this.word1.split("")
+			}
+			get bar() {
+				return this._bar
+			}
+			setWord2() {
+				let mix = [...this.chars].slice(0, 3)
+				mix[2] = "z"
+				this.word2 = mix.join("")
+			}
+			syncFoo() {
+				return produce(this, state => {
+					state.foo = state.bar.baz
+				})
+			}
+		}
+
+		const state = new State()
+
+		it("should work without creating a proxy for a getter property", () => {
+			expect(state.chars).toEqual(['b', 'a', 'r'])
+			const newState1 = produce(state, d => d.setWord2())
+			expect(newState1.word2).toEqual('baz')
+
+			expect(state.bar).toEqual({ baz: 1 })
+			const newState2 = state.syncFoo()
+			expect(newState2.foo).toEqual(1)
```

Comment:
I'd check here as well if `newState2.bar` output is correct (in a case were it differs from newState1), to verify that `this.` works everywhere as expected in those getters

Reply:
Done